Add glossary (#5935)
RadeonOpenCompute/ROCm: docs/reference/glossary/host-software.rst

    profiler, amd-smi

.. _glossary-host-software:

*************
Host software
*************

This section provides brief definitions of development tools, compilers,
libraries, and runtime environments for programming AMD GPUs.

.. glossary::
    :sorted:

    ROCm software platform
        ROCm is AMD's GPU software stack, providing compiler
        toolchains, runtime environments, and performance libraries for HPC and
        AI applications. See :doc:`../../what-is-rocm` for a complete component
        overview.

    HIP C++
        HIP C++ is an extension of C++ for writing portable GPU programs that
        target AMD GPUs. See :doc:`hip:how-to/hip_cpp_language_extensions` for
        language fundamentals.

    amd-smi
        The ``amd-smi`` command-line utility queries, monitors, and manages AMD GPU
        state, providing hardware information and performance metrics. See
        :doc:`amdsmi:index` for detailed usage.

    HIP runtime API
        The HIP runtime API provides a C-style interface for GPU programming, offering
        functions for memory management, kernel launches, and synchronization. See

>         :ref:`hip:hip_runtime_api_how-to` for API overview.

Reviewer: How is the HIP runtime API different from HIP C++?

Reviewer 2: Good catch replaced HIP C++ with HIP C++ language extension, please check it now.

    HIP compiler
        The HIP compiler ``amdclang++`` compiles HIP C++ programs into binaries
        containing both host CPU and device GPU code. See
        :doc:`llvm-project:reference/rocmcc` for compiler flags and options.

    HIP runtime compiler
        The HIP Runtime Compiler (HIPRTC) compiles HIP source code at runtime
        into AMDGPU binary code objects for just-in-time kernel generation. See

>         :ref:`hip:hip_runtime_compiler_how-to` for API details.

Reviewer: for use on different devices? Why compile at runtime?

Reviewer 2: Updated.

>     ROCm profiler

Reviewer: This is good. Should we have one for ROCgdb?

Reviewer 2: Good idea, added.

.. meta::
  :description: Host software glossary for AMD GPUs
  :keywords: AMD, ROCm, GPU, host software, HIP, compiler, runtime, libraries,
    profiler, amd-smi

.. _glossary-host-software:

*************

> Host software

Reviewer 3: ```suggestion
Host software glossary
```

*************

> 

Reviewer 3: Finding it strange that this glossary holds definitions for items in the What is ROCm topic. 
For example, AMD SMI

This section provides brief definitions of development tools, compilers,
libraries, and runtime environments for programming AMD GPUs.

.. glossary::
    :sorted:

    ROCm software platform
        ROCm is AMD's GPU software stack, providing compiler
        toolchains, runtime environments, and performance libraries for HPC and
        AI applications. See :doc:`../../what-is-rocm` for a complete component
        overview.

    HIP C++ language extension
        HIP extends the C++ language with additional features designed for
        programming heterogeneous applications. These extensions mostly relate
        to the kernel language, but some can also be applied to host
        functionality. See :doc:`hip:how-to/hip_cpp_language_extensions` for
        language fundamentals.

>     amd-smi

Reviewer 3: Why is this not matching the What is ROCm? Both the naming and definition:

<img width="1547" height="473" alt="Image" src="https://github.com/user-attachments/assets/63e3b6e6-4476-4e27-af32-25781eafdba5" />

        The ``amd-smi`` command-line utility queries, monitors, and manages AMD GPU
        state, providing hardware information and performance metrics. See
        :doc:`amdsmi:index` for detailed usage.

    HIP runtime API
        The HIP runtime API provides an interface for GPU programming, offering
        functions for memory management, kernel launches, and synchronization. See
        :ref:`hip:hip_runtime_api_how-to` for API overview.

>     HIP compiler
>         The HIP compiler ``amdclang++`` compiles HIP C++ programs into binaries
>         containing both host CPU and device GPU code. See
>         :doc:`llvm-project:reference/rocmcc` for compiler flags and options.

Reviewer 3: ```suggestion
    HIP compiler
        The HIP compiler ``amdclang++`` compiles HIP C++ programs into binaries
        that contain both host CPU and device GPU code. See
        :doc:`llvm-project:reference/rocmcc` for compiler flags and options.
```

    HIP runtime compiler
        The HIP Runtime Compiler (HIPRTC) compiles HIP source code at runtime

>         into AMDGPU binary code objects, enabling just-in-time kernel generation,

Reviewer 3: It would be nice to define "AMDGPU binary"

        device-specific optimization, and dynamic code creation for different
        GPUs. See :ref:`hip:hip_runtime_compiler_how-to` for API details.

    ROCgdb
        ROCgdb is AMD's source-level debugger for HIP and ROCm applications,
        enabling debugging of both host CPU and GPU device code, including
        kernel breakpoints, stepping, and variable inspection. See
        :doc:`rocgdb:index` for usage and command reference.

>     ROCm profiler

Reviewer 3: The title of this is inconsistent with the names in What is ROCm:

<img width="1520" height="875" alt="Image" src="https://github.com/user-attachments/assets/1e579e0a-7691-4d14-a561-8d4b450f0f12" />

    profiler, amd-smi

.. _glossary-host-software:

*************
Host software
*************

This section provides brief definitions of development tools, compilers,
libraries, and runtime environments for programming AMD GPUs.

.. glossary::
    :sorted:

    ROCm software platform
        ROCm is AMD's GPU software stack, providing compiler
        toolchains, runtime environments, and performance libraries for HPC and
        AI applications. See :doc:`../../what-is-rocm` for a complete component
        overview.

    HIP C++ language extension
        HIP extends the C++ language with additional features designed for
        programming heterogeneous applications. These extensions mostly relate
        to the kernel language, but some can also be applied to host
        functionality. See :doc:`hip:how-to/hip_cpp_language_extensions` for
        language fundamentals.

    amd-smi
        The ``amd-smi`` command-line utility queries, monitors, and manages AMD GPU
        state, providing hardware information and performance metrics. See
        :doc:`amdsmi:index` for detailed usage.

    HIP runtime API

>         The HIP runtime API provides a C-style interface for GPU programming, offering

Reviewer 2: ```suggestion
        The HIP runtime API provides an interface for GPU programming, offering
```

        functions for memory management, kernel launches, and synchronization. See
        :ref:`hip:hip_runtime_api_how-to` for API overview.

    HIP compiler
        The HIP compiler ``amdclang++`` compiles HIP C++ programs into binaries
        containing both host CPU and device GPU code. See
        :doc:`llvm-project:reference/rocmcc` for compiler flags and options.

> 
>     HIP runtime compiler
>         The HIP Runtime Compiler (HIPRTC) compiles HIP source code at runtime
>         into AMDGPU binary code objects for just-in-time kernel generation. See
>         :ref:`hip:hip_runtime_compiler_how-to` for API details.

Reviewer 2: ```suggestion
```
